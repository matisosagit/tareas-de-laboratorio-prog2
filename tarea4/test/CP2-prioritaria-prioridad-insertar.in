# Test para TColaDePrioridadPerros: prioritario, prioridad, insertar
crearCP 25

# creo Perro 1 (Vitalidad 10)
crearFecha 15/10/2022
crearPerro 1 Bruno 5 10 Un Beagle amigable
insertarEnCP
prioridadEnCP 1

# Perro 2 (Vitalidad 5) - más prioritario que Perro 1
crearFecha 10/10/2022
crearPerro 2 Luna 3 5 Una Labradora juguetona
insertarEnCP
prioridadEnCP 2

# el perro prioritario es el 2 (Vitalidad 5)
prioritarioCP

# creo Perro 3 (Vitalidad 10) - igual que Perro 1
crearFecha 15/10/2022
crearPerro 3 Coco 2 10 Un Caniche inteligente
insertarEnCP
prioridadEnCP 3

# el perro prioritario sigue siendo 2 (Vitalidad 5)
prioritarioCP

# Perro 1 está en la cola
estaEnCP 1

# creo Perro 4 (Vitalidad 3) e inserto en cola - nuevo prioritario
crearFecha 01/10/2022
crearPerro 4 Lola 1 3 Una Bulldog cariñosa
insertarEnCP
prioridadEnCP 4

# ahora el perro prioritario es el 4 (Vitalidad 3)
prioritarioCP

# creo Perro 5 (Vitalidad 1) como mas prioritario y agrego a cola
crearFecha 01/10/2020
crearPerro 5 Max 6 1 Un Pastor Alemán leal
insertarEnCP
prioridadEnCP 5

# creo Perro 6 (Vitalidad 8) y agrego a cola
crearFecha 07/10/2022
crearPerro 6 Kira 4 8 Una Husky energética
insertarEnCP
prioridadEnCP 6

# creo Perro 7 (Vitalidad 9) y agrego a cola
crearFecha 10/09/2022
crearPerro 7 Rocky 3 9 Un Boxer valiente
insertarEnCP
prioridadEnCP 7

# creo Perro 8 (Vitalidad 12) y agrego a cola
crearFecha 12/10/2022
crearPerro 8 Toby 5 12 Un perro adorable
insertarEnCP
prioridadEnCP 8

# creo Perro 9 (Vitalidad 4) y agrego a cola
crearFecha 02/10/2022
crearPerro 9 Nala 2 4 Una Golden Retriever dócil
insertarEnCP
prioridadEnCP 9

# creo Perro 10 (Vitalidad 7) y agrego a cola
crearFecha 09/10/2022
crearPerro 10 Simba 1 7 Un Chihuahua pequeño pero audaz
insertarEnCP
prioridadEnCP 10

# creo Perro 11 (Vitalidad 6) y agrego a cola
crearFecha 03/10/2022
crearPerro 11 Mia 4 6 Una Poodle elegante
insertarEnCP
prioridadEnCP 11

# creo Perro 12 (Vitalidad 5) y agrego a cola - igual vitalidad que Perro 2
crearFecha 04/10/2022
crearPerro 12 Zeus 3 5 Un Doberman protector
insertarEnCP
prioridadEnCP 12

# ahora el perro prioritario es el 5 (Vitalidad 1)
prioritarioCP

# verifico que el perro 8 está en la cola
estaEnCP 8

# verifico que el perro 7 está en la cola
estaEnCP 7

liberarCP

Fin