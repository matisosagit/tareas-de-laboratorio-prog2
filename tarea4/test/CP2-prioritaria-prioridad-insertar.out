1> #  Test para TColaDePrioridadPerros: prioritario, prioridad, insertar.
2> Se creó una cola de prioridad de perros con cantidad maxima 25
3> #  creo Perro 1 (Vitalidad 10).
4> Fecha creada en forma exitosa.
5> 6> Se insertó el perro en la cola de prioridad.
7> La prioridad del perro de id 1 es 10
8> #  Perro 2 (Vitalidad 5) - más prioritario que Perro 1.
9> Fecha creada en forma exitosa.
10> 11> Se insertó el perro en la cola de prioridad.
12> La prioridad del perro de id 2 es 5
13> #  el perro prioritario es el 2 (Vitalidad 5).
14> Perro 2
Nombre: Luna
Edad: 3
Descripcion:  Una Labradora juguetona
Fecha de ingreso: 10/10/2022
Vitalidad: 5
15> #  creo Perro 3 (Vitalidad 10) - igual que Perro 1.
16> Fecha creada en forma exitosa.
17> 18> Se insertó el perro en la cola de prioridad.
19> La prioridad del perro de id 3 es 10
20> #  el perro prioritario sigue siendo 2 (Vitalidad 5).
21> Perro 2
Nombre: Luna
Edad: 3
Descripcion:  Una Labradora juguetona
Fecha de ingreso: 10/10/2022
Vitalidad: 5
22> #  Perro 1 está en la cola.
23> El perro con id 1 está en la cola de prioridad.
24> #  creo Perro 4 (Vitalidad 3) e inserto en cola - nuevo prioritario.
25> Fecha creada en forma exitosa.
26> 27> Se insertó el perro en la cola de prioridad.
28> La prioridad del perro de id 4 es 3
29> #  ahora el perro prioritario es el 4 (Vitalidad 3).
30> Perro 4
Nombre: Lola
Edad: 1
Descripcion:  Una Bulldog cariñosa
Fecha de ingreso: 1/10/2022
Vitalidad: 3
31> #  creo Perro 5 (Vitalidad 1) como mas prioritario y agrego a cola.
32> Fecha creada en forma exitosa.
33> 34> Se insertó el perro en la cola de prioridad.
35> La prioridad del perro de id 5 es 1
36> #  creo Perro 6 (Vitalidad 8) y agrego a cola.
37> Fecha creada en forma exitosa.
38> 39> Se insertó el perro en la cola de prioridad.
40> La prioridad del perro de id 6 es 8
41> #  creo Perro 7 (Vitalidad 9) y agrego a cola.
42> Fecha creada en forma exitosa.
43> 44> Se insertó el perro en la cola de prioridad.
45> La prioridad del perro de id 7 es 9
46> #  creo Perro 8 (Vitalidad 12) y agrego a cola.
47> Fecha creada en forma exitosa.
48> 49> Se insertó el perro en la cola de prioridad.
50> La prioridad del perro de id 8 es 12
51> #  creo Perro 9 (Vitalidad 4) y agrego a cola.
52> Fecha creada en forma exitosa.
53> 54> Se insertó el perro en la cola de prioridad.
55> La prioridad del perro de id 9 es 4
56> #  creo Perro 10 (Vitalidad 7) y agrego a cola.
57> Fecha creada en forma exitosa.
58> 59> Se insertó el perro en la cola de prioridad.
60> La prioridad del perro de id 10 es 7
61> #  creo Perro 11 (Vitalidad 6) y agrego a cola.
62> Fecha creada en forma exitosa.
63> 64> Se insertó el perro en la cola de prioridad.
65> La prioridad del perro de id 11 es 6
66> #  creo Perro 12 (Vitalidad 5) y agrego a cola - igual vitalidad que Perro 2.
67> Fecha creada en forma exitosa.
68> 69> Se insertó el perro en la cola de prioridad.
70> La prioridad del perro de id 12 es 5
71> #  ahora el perro prioritario es el 5 (Vitalidad 1).
72> Perro 5
Nombre: Max
Edad: 6
Descripcion:  Un Pastor Alemán leal
Fecha de ingreso: 1/10/2020
Vitalidad: 1
73> #  verifico que el perro 8 está en la cola.
74> El perro con id 8 está en la cola de prioridad.
75> #  verifico que el perro 7 está en la cola.
76> El perro con id 7 está en la cola de prioridad.
77> Se liberó la cola de prioridad de perros
78> Fin.
